packages/nextjs-mf/src/plugins/JsonpChunkLoading.ts

```diff
@@ -0,0 +1,314 @@
+            'webpackJsonpCallback.apply(',
+            Template.indent([
+              'null,',
+              '[null].concat(Array.prototype.slice.call(arguments))',
+            ]),
+            ');',
+            'return originalPush.apply(chunkLoadingGlobal, arguments);',
+          ]),
+          '}',
+          // Template.indent([
+          //   "console.log('current push args', arguments[0][0]);",
+          //   "console.log('whole queue', chunkQueue);",
+          //   "console.log('report queue', resport);",
+          // ]),
+          "console.log('pushing chunk into webpack runtime:', arguments[0][0]);",
+          'webpackJsonpCallback.apply(',
+          Template.indent([
+            'null,',
+            '[null].concat(Array.prototype.slice.call(arguments))',
+          ]),
+          ');',
+          'return originalPush.apply(chunkLoadingGlobal, arguments);',
+        ]),
+        '};',
+      ]),
+      '})(chunkLoadingGlobal.push.bind(chunkLoadingGlobal));',
+    ]),
+  ];
+
+  return Template.asString(code);
+}
+
+function buildOutputString(dependencies: Record<string, any>): string[] {
+  const output: string[] = [];
+  for (const [key, version] of Object.entries(dependencies)) {
+    output.push(
+      `__webpack_require__.S.default[${JSON.stringify(key)}][${JSON.stringify(
+        version
+      )}]`
+    );
+  }
+  return output;
+}
+
+class CustomWebpackPlugin {
+  private initialModules: Set<any>;
+  private initialModulesFound: Set<any>;
+
+  constructor() {
+    this.initialModules = new Set();
+    this.initialModulesFound = new Set();
+  }
+
+  apply(compiler: Compiler): void {
+    compiler.hooks.compilation.tap(
+      'CustomWebpackPlugin',
+      (compilation: Compilation) => {
+        const addModules = (modules: Iterable<Module>, chunk: Chunk): void => {
+          for (const module of modules) {
+            // const id =
+            //   compilation.chunkGraph.getModuleId(module) ||
+            //   module.id ||
+            //   module.debugId;
+            this.initialModules.add(module);
+            // @ts-ignore
+            this.initialModulesFound.add(module?.options?.shareKey);
+          }
+        };
+
+        compilation.hooks.optimizeChunks.tap(
+          'AddModulesToRuntimeChunkPlugin',
+          (chunks) => {
+            for (const entrypointModule of compilation.entrypoints.values()) {
+              const entrypoint = entrypointModule.getEntrypointChunk();
+              if (entrypoint.name === compiler.options.output.uniqueName)
+                continue;
+
+              const processChunks = (
+                chunks: Set<Chunk>,
+                callback: (modules: Iterable<Module>, chunk: any) => void
+              ): void => {
+                for (const chunk of chunks) {
+                  const modules =
+                    compilation.chunkGraph.getChunkModulesIterableBySourceType(
+                      chunk,
+                      'consume-shared'
```

Comment:
Get any module types in chunk that match consume-shared, these are eager shared since they are referenced in the entrypoint.